### ISSUE TITLE
possibility in using Haxe?

### ISSUE THREAD

**Issue by hyusetiawan** [aichaos/rivescript-python#23]
i see that rivescript has a lot of client support, have you considered using Haxe as the single parser and export to different clients? The language is based off of ECMA / AS3 and designed to be translated to other languages: http://haxe.org/

**Comment by kirsle** [comment]
I hadn't heard of Haxe before. It looks like Haxe would be another rewrite of RiveScript (in Haxe language) that would be exported out as JavaScript/Python/C++/other languages they support instead of having to have implemented RiveScript in each language directly?

**Comment by hyusetiawan** [comment]
that is correct, I am not affiliated with Haxe (only used it once for cross platform mobile apps), the language exports out a sane idiomatic translated language as well. Just a suggestion to minimize supported client work :)

PS: sorry for forgetting to label this as a question :P

**Comment by kirsle** [comment]
That's alright (in the label department); I thought this was an issue on `rivescript-js` at first, which gets more issues than the other ones and I have more labels to categorize them all.

After the Go port of RiveScript I decided I was tired of reimplementing RiveScript and probably won't do so again. Go can currently cross-compile into JavaScript with gopherjs (though probably not as efficiently as writing native JS directly :wink:), and they're working on C-compatible shared objects to make it linkable with most C-derived languages. So if I wanna bring RiveScript to a new language in the future I'd probably go for the C-API route with the Go version.

**Comment by hyusetiawan** [comment]
+1
does rivescript have a community chat? It's ironic a chat bot language does not have a chat channel hah
Slack perhaps? I only know of https://www.chatbots.org/ai_zone/viewforum/48/

**Comment by kirsle** [comment]
The forum on Chatbots.org is the main place, yeah. I used to manage my own [YaBB forum](http://www.yabbforum.com/), but it wasn't frequently posted to (apart from spam bots), so I delegated the responsibility of running a forum to Chatbots.org.

I don't have my own Slack group for it. I do hang out on the [Botmakers Slack](https://botmakers.org/), which is a fairly active community of all kinds of botmakers and includes a handful of people who contributed to RiveScript (@dcsan, @gleuch), and Rob Ellis (@silentrob; creator of the [SuperScript](https://github.com/superscriptjs/superscript) fork of RiveScript).

Rob has his own Slack group specifically for SuperScript, but I think a RiveScript-specific Slack group would have more tumbleweeds in it than users :wink: -- I don't market RiveScript as well as Rob markets SuperScript.
